### ISSUE TITLE
[question/feature] Where is the MC10 code located by default?

### ISSUE THREAD

**Issue by Fabrizio-Caruso** [EtchedPixels/CC6303#20]
Where is the code for the MC10 target located by default when compiled by running the following command?
```
cc68 -tmc10 foo.c -o foo
```
Is it located at $4200 just after text mode video ram?
I am asking because I fear that this would break any program that uses graphics mode (as they may take 3k or $4000-$4BFFF) if I am doing the math correctly.

I have tried to use the `-C` and `--start-addr` options with `cc68` but they are not available. 
I have not found a cfg linker file to control where the sections are in memory (such as the ones in CC65).

I see there may be a way to do this by using the linker `ld68` command, though.

**Comment by Fabrizio-Caruso** [comment]
Thanks for replying!
"Hires" 4 color 128x96 graphics takes $4000-$4BFF (19455). So a more general default would be $4C00 (19456), because it would allow all programs to run at the cost of memory when no graphics is used.
Strangely I can have a loop writing into $4000-$4BFF without a crash/freeze.
I am compiling with the cc68 but I see that I should do it with the linker to avoid such problems.
Is there a way to the high level command cc68 to start at $4C00 (so that programs with graphics would run without a crashing)? Would you please consider adding an option for this in the future?

**Comment by Fabrizio-Caruso** [comment]
@EtchedPixels Did you mean 17500 or 19500? Sorry for asking again.
I ask because I can have a loop that writes from $4000 through $4BFF (19455) without the program crashing/freezing.

**Comment by Fabrizio-Caruso** [comment]
If the top level cc68 command does not have an option for the start-address, would you please consider having such an option to pass it to the linker for the sake of convenience? If that is not possible or too much work, would you consider $4C00 as default?

For my specific case I can rewrite my Makefile to use the explicit compile only cc68, the linker command and the tool to generate a tape image as separate commands.

**Comment by EtchedPixels** [comment]
17500.. it's the right start spot for most MC-10 code, especially as you've usually on got 4K of RAM.

I'll look at a setting to change the code base or a variant MC10 build type.

**Comment by Fabrizio-Caruso** [comment]
@EtchedPixels I understand. If a setting is too much of a hassle, a second target, e.g mc10exp, with start address at $4C00 (19456) should also do because $0C00=3k=128x96 in 4 colors, which is the highest possible resolution on an unmodified M-10. 
For the unexpanded mc10, why exactly 17500? If my math is correct, $4200 (16896) would be enough for the default screen text mode and provide more memory for the code. Plain text mode should only take 32x16=512 ($0200) bytes. 
So my guess is that maybe you wanted to leave 1k for the super low res graphics 64x64 in 4 colors.

**Comment by EtchedPixels** [comment]
17500 was what I was told to use - I assume there's a bit below it still being used by basic and system variables on the load/exec.

**Comment by Fabrizio-Caruso** [comment]
@EtchedPixels I am no expert. I would like to run some experiments with lower and higher start address. Do I only need to change all occurrences of 17500 in cc.c or is 17500 also coded elsewhere?

**Comment by EtchedPixels** [comment]
I'm planning to retire this compiler in favour of the one I've been working on from scratch so don't intend to do anything to it at this point. Once that is stable I'll have a look at splitting out the other tools from this one for the MC10 so it's easier to use including default addresses etc.

**Comment by Fabrizio-Caruso** [comment]
@EtchedPixels Is the forked version of this repository the new compiler you refer to?
I have opened the very same feature request there for the mc10 target, i.e. request for an option to set the start address without the need to modify the compiler code.

**Comment by EtchedPixels** [comment]
No. It's unrelated. The new compiler is the one I built mostly to do all the Fuzix stuff. The assembler and linker are the same however so the same options for setting addresses apply.

Note btw - I've moved from Github so may not often see anything here.

**Comment by Fabrizio-Caruso** [comment]
@EtchedPixels Sorry for not being clear. My bad.
I am using CC6303 with the `-t mc10` option. This option is very convenient because it does everything (it uses the correct crt) and produces a ready to use tape file for the mc10 emulators but has the start address hardcoded.

It would be of great help to have `-t mc10ext` to just do the same thing but with a start address above the screen ram. 
Screen ram is $4000-$4BFF (19455). So, for the expanded mc10, a more useful default would be $4C00 (19456).

I am not sure I know which
